test/parallel/test-tick-processor.js

```diff
@@ -0,0 +1,38 @@
+'use strict';
+var fs = require('fs');
+var assert = require('assert');
+var path = require('path');
+var cp = require('child_process');
+var common = require('../common');
+
+common.refreshTmpDir();
+
+process.chdir(common.tmpDir);
+cp.execFileSync(process.execPath, ['-prof', '-pe',
+    'function fib(n) { require(\'vm\').runInDebugContext(\'Debug\');' +
+    'return n < 2 ? n : setImmediate(function() { fib(n-1) + fib(n-2);}); };' +
+    'setTimeout(function() { process.exit(0); }, 2000);' +
+    'fib(40);']);
```

Comment:
I don't think this is a reliable way to detect for the presence of RunInDebugContext.  The V8 profiler is a sampling profiler, running at a 1 kHz sampling frequency.  It could easily miss the call to runInDebugContext().

What should work (I think) is to run the fib function _inside_ the debug context, that way there should always be a RunInDebugContext stack frame on the stack.

Apropos the setTimeout call, is that to kill the script when it takes too long to complete?  That won't work for synchronous code like fib().